src/plugins/embeddable/public/lib/attribute_service/attribute_service.tsx

```diff
@@ -169,8 +169,9 @@ export class AttributeService<
           // Remove unneeded attributes from the original input.
           const newInput = omit(input, ATTRIBUTE_SERVICE_KEY);
 
-          // Combine input and wrapped input to preserve any passed in explicit Input.
-          resolve({ ...newInput, ...wrappedInput });
+          // Combine input and wrapped input to preserve any passed in explicit Input while ensuring that the
+          // library title ovewrites the original title
+          resolve({ ...newInput, ...wrappedInput, title: newAttributes.title });
```

Comment:
nit; This code is a little tricksy to follow 😅 it seems a little clearer to me if we did:

```ts
          const wrappedInput = (await this.wrapAttributes(newAttributes, true, {
            title: props.newTitle,
          })) as RefType;
```

perhaps @Dosant has some input (pun intended) too.